homework05/test/examples/Main.hs

```diff
@@ -1,15 +1,63 @@
-module Main
-  ( main
-  )
-  where
-
--- doctest
-import qualified Test.DocTest as DocTest
+module Main 
+  ( main )
+where
 
+import ExprT
+import Calc
+import Parser
+import StackVM
+import Test.Hspec
 
 main :: IO ()
-main =
-  DocTest.doctest
-    [ "-isrc"
-    , "src/Calc.hs"
-    ]
+main = hspec $ do
+  
+  describe "eval" $
+    it "returns result of applying add and multiply operations" $
+      eval (ExprT.Mul (ExprT.Add (ExprT.Lit 2) (ExprT.Lit 3)) (ExprT.Lit (-4))) `shouldBe` (-20)
+
+  describe "evalStr" $ do
+    it "evaluates expression with parenthesis" $
+      evalStr "(2+3)*(-4)" `shouldBe` Just (-20)
+
+    it "evaluates expression without parenthesis" $
+      evalStr "2+3*(-4)" `shouldBe` Just (-10)
+
+    it "evaluates wrong expression" $
+      evalStr "2+3*" `shouldBe` Nothing
+      
+  describe "reify" $
+    it "makes an instance of Expr to ExprT type" $
+      reify (mul (add (lit 2) (lit 3)) (lit (-4))) 
+        `shouldBe` ExprT.Mul (ExprT.Add (ExprT.Lit 2) (ExprT.Lit 3)) (ExprT.Lit (-4))
+  
+  describe "parseExp" $
+    context "when provided with multiple data types" $ do
```

Comment:
@jmorocho I think it's better to have multiple contexts here. For example: 

```haskell
context "when provided with an Integer data type"
  it "evaluates the expression"
```

What do you think?